plugins/tech-insights-backend/src/types.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { DateTime } from 'luxon';
+import {
+  DynamicFactCallback,
+  Event,
+  FactOptions,
+  TopLevelCondition,
+} from 'json-rules-engine';
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+
+export type TechInsightFact = {
+  ref: string;
+  entity: {
+    namespace: string;
+    kind: string;
+    name: string;
+  };
+  facts: Record<string, number | string | boolean | DateTime>;
+};
+
+type FactValueDefinitions = {
+  [key: string]: {
+    type: 'integer' | 'float' | 'string' | 'boolean' | 'datetime';
+    description?: string;
```

Comment:
First of all great work with this PRFC! 🙌 

Thinking ahead when we want to present things in the UI, wouldn't it be good to require a descriptions from the start in all cases or will we rely on information from some other place to make sense of the check?

I might be confusing the terms but I just want to make sure that we make sure to make checks actionable for users somehow.
For example;
As I user visiting my component in backstage I see a check saying "No PD schedule added". There should be able to get sufficient additional information to take action and fix that check, like toggling some additional description or clicking a "fix me" link that takes me to PD or tech docs.

Reply:
Going back to the diagram of this PRFC, I see that checks are a runtime concern. So if I would try to address my previous comment would the "actionable" part be something I solve myself in the UI bits when using these plugin(s)? :)

Reply:
Yeah, I think in the end the handling of a check result will need to be defined on the check. There isn't much on that yet within this PR. [I'm just piping through whatever `json-rules-engine` pushes out at the moment](https://github.com/backstage/backstage/pull/7193/files#diff-9214eef2996f2ab9b4758817467f438a60e752bfc9613ffb2d62dea2445fd7cbR51), so the actionable item type is still very much undecided. I feel we need to create a set of types that define different types of actions (link, something else?) that have needed properties and which are returned as part of the check result. FE can then display them based on those types to make them actionable on the UI side indeed.